scripts/corr2_bengine_status.py

```diff
@@ -0,0 +1,168 @@
+                  'register %s' % (fpga_.host, 'bf%i_out_count' % ctr)
+            signal_handler(None, None)
+        if 'bf%i_ot_count' % ctr not in fpga_.registers.names():
+            print 'ERROR: host %s does not have necessary beamformer ' \
+                  'register %s' % (fpga_.host, 'bf%i_of_count' % ctr)
+            signal_handler(None, None)
+
+
+def get_fpga_data(fpga_):
+    rv = {}
+    for ctr in range(num_beams):
+        rvt = {'out': fpga_.registers['bf%i_out_count' % ctr].read()['data'],
+               'of': fpga_.registers['bf%i_of_count' % ctr].read()['data']}
+        rv[ctr] = rvt
+    return rv
+
+headers = ['out%i' % ctr for ctr in range(4)]
+headers.extend(['of%i' % ctr for ctr in range(4)])
+
+# set up the curses scroll screen
+scroller = scroll.Scroll(debug=False)
+scroller.screen_setup()
+# main program loop
+STARTTIME = time.time()
+last_refresh = STARTTIME - 3
+try:
+    while True:
+        # get key presses from ncurses
+        keypress = scroller.on_keypress()[0]
+        if keypress == -1:
+            break
+        elif keypress > 0:
+            scroller.draw_screen()
+        if time.time() > last_refresh + polltime:
+            scroller.clear_buffer()
+            scroller.add_line('Polling %i x-engine%s every %s - %is elapsed.' %
+                              (len(fpgas), '' if len(fpgas) == 1 else 's',
+                               'second' if polltime == 1 else ('%i seconds' % polltime),
+                               time.time() - STARTTIME), 0, 0, absolute=True)
+#            scroller.add_line('hdr1', 10)
+#            scroller.add_line('hdr2', 30, scroller.get_current_line())
+            scroller.add_line('Host - out(0,1,2,3) of(0,1,2,3)',
+                              0, 1, absolute=True)
+            scroller.set_ypos(2)
+            scroller.set_ylimits(ymin=2)
+            all_fpga_data = fpgautils.threaded_fpga_operation(
+                fpgas, 10, get_fpga_data)
+            for ctr, fpga_ in enumerate(fpgas):
+                fpga_data = all_fpga_data[fpga_.host]
+                scroller.add_line(fpga_.host)
+                for beam_num, beam_data in fpga_data.items():
+                    beamd = fpga_data[beam_num]
+                    scroller.add_line('beam%i%10i%10i%10i%10i%10i%10i%10i%10i'
+                                      % (beam_num,
+                                         beamd['out']['count0'],
+                                         beamd['out']['count1'],
+                                         beamd['out']['count2'],
+                                         beamd['out']['count3'],
+                                         beamd['of']['count0'],
+                                         beamd['of']['count1'],
+                                         beamd['of']['count2'],
+                                         beamd['of']['count3']), 5)
+            scroller.draw_screen()
+            last_refresh = time.time()
+except Exception, e:
```

Comment:
No need for the `, e`, since you are not using this instance. Also, do you want signal_handler to be called for Ctrl-C and/or kill 15? In that case you should add `KeyboardInterrupt` and/or `SystemExit` (or some such, not 100% sure), or even just use a *shudder* naked except.